src/qnetvo/network_ansatz.py

```diff
@@ -283,64 +319,56 @@ def circuit(settings):
+        rand_settings = [
+            np.array(2 * np.pi * np.random.rand() - np.pi) for _ in range(num_settings)
         ]
+        if len(fixed_setting_ids) > 0 and len(fixed_settings) > 0:
+            for i, id in enumerate(fixed_setting_ids):
+                rand_settings[id] = np.array(fixed_settings[i], requires_grad=False)
 
-        return [prepare_settings, measure_settings]
+        return rand_settings
 
-    def tf_rand_scenario_settings(self):
+    def tf_rand_network_settings(self, fixed_setting_ids=[], fixed_settings=[]):
         """Creates a randomized settings array for the network ansatz using TensorFlow
         tensor types.
 
-        :returns: See :meth:`qnetvo.NetworkAnsatz.rand_scenario_settings` for details.
-        :rtype: list[list[tf.Tensor]]
+        :param fixed_setting_ids: The ids of settings that are held constant during optimization.
+        :type fixed_setting_ids: *optional* List[Int]
+
+        :param fixed_settings: The constant values for fixed settings.
+        :type fixed_settings: *optional* List[Float]
+
+        :returns: See :meth:`qnetvo.NetworkAnsatz.rand_network_settings` for details.
+        :rtype: List[tf.Tensor]
         """
         from .lazy_tensorflow_import import tensorflow as tf
 
-        np_settings = self.rand_scenario_settings()
-
+        np_settings = self.rand_network_settings(fixed_setting_ids, fixed_settings)
         return [
-            [tf.Variable(settings) for settings in np_settings[0]],
-            [tf.Variable(settings) for settings in np_settings[1]],
+            tf.Variable(setting) if qml.math.requires_grad(setting) else tf.constant(setting)
+            for setting in np_settings
         ]
 
-    def zero_scenario_settings(self):
+    def zero_network_settings(self):
         """Creates a settings array for the network ansatz that consists of zeros.
 
-        :returns: A nested list containing settings for each network node.
-                  ``PrepareNode`` settings are listed under index ``0`` while
-                  ``MeasureNode`` settings are listed under index ``1``.
-                  The measure and prepare layers settings are a list of numpy arrays
-                  where the dimension of each array is ``(num_inputs, num_settings)``.
-        :rtype: list[list[np.array]]
+        :returns: A 1D list of ``np.tensor`` scalar values having ``requires_grad=True``.
+        :rtype: List[Float]
         """
-        prepare_settings = [
-            np.zeros(node.settings_dims) if len(node.static_settings) == 0 else np.array([[]])
-            for node in self.prepare_nodes
-        ]
-        measure_settings = [
-            np.zeros(node.settings_dims) if len(node.static_settings) == 0 else np.array([[]])
-            for node in self.measure_nodes
-        ]
-
-        return [prepare_settings, measure_settings]
+        num_settings = self.parameter_partitions[-1][-1][-1][-1]
+        return [np.array(0) for _ in range(num_settings)]
```

Comment:
```suggestion
        return [np.array(0, requires_grad=True) for _ in range(num_settings)]
```
Maybe better to be explicit. I recall discussing the merits of having `requires_grad=False` by default in PL.